src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LanguageServerProjectSystem.cs

```diff
@@ -255,6 +273,8 @@ private async Task<(LSP.MessageType? FailureType, BuildHostProcessKind? Preferre
 
                         projectFileInfos[loadedProjectInfo] = await loadedProject.UpdateWithNewProjectInfoAsync(loadedProjectInfo, _logger);
                     }
+
+                    hasUnresolvedDependencies = ProjectDependencyHelper.HasUnresolvedDependencies(loadedProjectInfo, _logger);
```

Comment:
```suggestion
                    hasUnresolvedDependencies ||= ProjectDependencyHelper.HasUnresolvedDependencies(loadedProjectInfo, _logger);
```

Since there might be a conditional dependency in a specific TFM only?